lib/galaxy/web/security/__init__.py

```diff
@@ -11,6 +11,11 @@
 
 log = logging.getLogger(__name__)
 
+MAXIMUM_ID_SECRET_BITS = 448
+MAXIMUM_ID_SECRET_LENGTH = MAXIMUM_ID_SECRET_BITS / 8
```

Comment:
Is this guaranteed? 1 byte per character? Always ascii? In the [sample](https://github.com/galaxyproject/galaxy/blob/dev/config/galaxy.ini.sample#L980) we say 'any string'

Reply:
well it isn't going to be smaller than this right? I'll try an unpatriotic string and see what happens.

Reply:
Well - I guess the code is correct the docs are wrong:

```
======================================================================
ERROR: test_security_helper.test_maximum_length_handling_nonascii
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/john/.pyenv/versions/2.7.12/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/Users/john/workspace/galaxy/test/unit/test_security_helper.py", line 46, in test_maximum_length_handling_nonascii
    helper = security.SecurityHelper(id_secret=longest_id_secret)
  File "/Users/john/workspace/galaxy/lib/galaxy/web/security/__init__.py", line 49, in __init__
    self.id_cipher = Blowfish.new(self.id_secret)
  File "/Users/john/.pyenv/versions/2.7.12/lib/python2.7/site-packages/Crypto/Cipher/Blowfish.py", line 101, in new
    return BlowfishCipher(key, *args, **kwargs)
  File "/Users/john/.pyenv/versions/2.7.12/lib/python2.7/site-packages/Crypto/Cipher/Blowfish.py", line 65, in __init__
    blockalgo.BlockAlgo.__init__(self, _Blowfish, key, *args, **kwargs)
  File "/Users/john/.pyenv/versions/2.7.12/lib/python2.7/site-packages/Crypto/Cipher/blockalgo.py", line 141, in __init__
    self._cipher = factory.new(key, *args, **kwargs)
UnicodeEncodeError: 'ascii' codec can't encode characters in position 0-55: ordinal not in range(128)
```

Reply:
Pushed a clarification of this into galaxy.ini.sample in this branch - thanks for catch @martenson.

Reply:
I do not think that is correct though. I can use `č` as my `secret_id` and Galaxy runs fine.

Reply:
Hmm... but my test...

Reply:
Fine you win, it seems if you have a non-ascii character the maximum length is more like 448 / 24. I pulled out language describing that - since I'm not sure I understand it myself. But I do have test cases that clarify this is the boundary and that the ``_longest_bits`` is working properly even with long (though not as long) non-ASCII strings.

Reply:
Or is this just because my test is wrong:

```
>>> "◎" * (24 / 3 + 1)
'\xe2\x97\x8e\xe2\x97\x8e\xe2\x97\x8e\xe2\x97\x8e\xe2\x97\x8e\xe2\x97\x8e\xe2\x97\x8e\xe2\x97\x8e\xe2\x97\x8e'
```

Ugh...

Reply:
Updated the test case again. I think everything in here is right - it just unicode handling under Python 2 is a bit garbage. Having these test cases will hopefully help us when migrating though.